src/main/java/core/basesyntax/strategy/handler/ReturnOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public ReturnOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer currentQuantity = fruitDao.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (purchaseQuantity < 0) {
+            throw new IllegalArgumentException("Invalid return quantity - " + purchaseQuantity);
+        }
+        fruitDao.getStorage().put(transaction.getFruit(), currentQuantity + purchaseQuantity);
```

Comment:
Direct manipulation of the storage map is a violation of encapsulation. Consider adding a method in the FruitDao to handle the updating of quantities, rather than accessing the map directly.